HW_9/hw_9_2.js

```diff
@@ -0,0 +1,239 @@
+'use strict';
+/*
+1. Создайте функцию, принимающую число n, которая при каждом вызове будет генерировать случайное число [1 - n] включительно. 
+   Условие - каждый следующий вызов должен давать новое число (не встречавшееся в прошлых вызовах). 
+   И так пока не переберутся все n чисел. На n + 1 вызов и далее функция должна возвращать 'All numbers were received'. 
+   Задачу решить через замыкания
+    Например n = 5, функция выведет 5 чисел 1-5, а после будет выводить сугубо 'All numbers were received'
+*/
+
+const randomGenerator = (n) => {
+  const generatedNumbers = [];
+  let number;
+  return () => {
+    if (generatedNumbers.length === n) {
+      console.log('All numbers were received');
+    } else {
+      do {
+        number = Math.floor(Math.random() * n + 1);
+      } while (generatedNumbers.find(element => element === number));
```

Comment:
```suggestion
      } while (generatedNumbers.includes(number));
```
You are working with a number, includes will work)